cmd/agent/app/check.go

```diff
@@ -48,9 +63,33 @@ func init() {
 	checkCmd.Flags().IntVarP(&checkTimes, "check-times", "t", 1, "number of times to run the check")
 	checkCmd.Flags().IntVar(&checkPause, "pause", 0, "pause between multiple runs of the check, in milliseconds")
 	checkCmd.Flags().StringVarP(&logLevel, "log-level", "l", "", "set the log level (default 'off')")
-	checkCmd.Flags().IntVarP(&checkDelay, "delay", "d", 100, "delay between running the check and grabbing the metrics in miliseconds")
+	checkCmd.Flags().IntVarP(&checkDelay, "delay", "d", 100, "delay between running the check and grabbing the metrics in milliseconds")
 	checkCmd.Flags().BoolVarP(&formatJSON, "json", "", false, "format aggregator and check runner output as json")
 	checkCmd.Flags().StringVarP(&breakPoint, "breakpoint", "b", "", "set a breakpoint at a particular line number (Python checks only)")
+	checkCmd.Flags().BoolVarP(&profileMemory, "profile-memory", "m", false, "run the memory profiler (Python checks only)")
+
+	// Power user flags - mark as hidden
+	checkCmd.Flags().StringVar(&profileMemoryDir, "m-dir", "", "an existing directory in which to store memory profiling data, ignoring clean-up")
+	checkCmd.Flags().StringVar(&profileMemoryFrames, "m-frames", "", "the number of stack frames to consider")
+	checkCmd.Flags().StringVar(&profileMemoryGC, "m-gc", "", "whether or not to run the garbage collector to remove noise")
```

Comment:
Is there a reason you didn't use `BoolVar` for boolean flags?

Reply:
I did it so we would not have to keep defaults in sync in both places